xcdat/bounds.py

```diff
@@ -231,25 +256,41 @@ def get_bounds(
 
         return bounds
 
-    def add_bounds(self, axis: CFAxisKey) -> xr.Dataset:
+    def add_bounds(
+        self,
+        axis: CFAxisKey,
+        time_freq: Optional[Literal["year", "month", "day", "hour"]] = None,
+        daily_time_freq: Optional[Literal[1, 2, 3, 4, 6, 8, 12, 24]] = None,
+    ) -> xr.Dataset:
         """Add bounds for an axis using its coordinate points.
 
         This method loops over the axis's coordinate variables and attempts to
-        add bounds for each of them if they don't exist. The coordinates must
-        meet the following criteria in order to add bounds:
-
-          1. The axis for the coordinates are "X", "Y", "T", or "Z"
-          2. Coordinates are single dimensional, not multidimensional
-          3. Coordinates are a length > 1 (not singleton)
-          4. Bounds must not already exist.
-             * Determined by attempting to map the coordinate variable's
-             "bounds" attr (if set) to the bounds data variable of the same key.
-          5. Time axes should be composed of `cftime` objects.
```

Comment:
In addition to `cftime` objects, we should also support time coordinates composed of `datetime` objects. I removed #5 from the criteria.

Reply:
I added logic in the latest commit to support datetime objects.